util/proto/stream.go

```diff
@@ -0,0 +1,350 @@
+	}
+
+	return size
+}
+
+// RepeatedUInt64Marshal implements Stream interface.
+func (s *stream) RepeatedUInt64Marshal(field int, v []uint64) (int, error) {
+	if len(v) == 0 {
+		return 0, nil
+	}
+
+	prefix := field<<3 | 0x02
+	offset, err := s.WriteUvarint(uint64(prefix))
+	if err != nil {
+		return offset, err
+	}
+
+	_, arrSize := RepeatedUInt64Size(field, v)
+	off, err := s.WriteUvarint(uint64(arrSize))
+	offset += off
+	if err != nil {
+		return offset, err
+	}
+
+	for i := range v {
+		off, err := s.WriteUvarint(v[i])
+		offset += off
+		if err != nil {
+			return offset, err
+		}
+	}
+
+	return offset, nil
+}
+
+// RepeatedInt64Marshal implements Stream interface.
+func (s *stream) RepeatedInt64Marshal(field int, v []int64) (int, error) {
+	if len(v) == 0 {
+		return 0, nil
+	}
+
+	prefix := field<<3 | 0x02
+	offset, err := s.WriteUvarint(uint64(prefix))
+	if err != nil {
+		return offset, err
+	}
+
+	_, arrSize := RepeatedInt64Size(field, v)
+	off, err := s.WriteUvarint(uint64(arrSize))
+	offset += off
+	if err != nil {
+		return offset, err
+	}
+
+	for i := range v {
+		off, err := s.WriteUvarint(uint64(v[i]))
+		offset += off
+		if err != nil {
+			return offset, err
+		}
+	}
+
+	return offset, nil
+}
+
+// RepeatedUInt32Marshal implements Stream interface.
+func (s *stream) RepeatedUInt32Marshal(field int, v []uint32) (int, error) {
+	if len(v) == 0 {
+		return 0, nil
+	}
+
+	prefix := field<<3 | 0x02
+	offset, err := s.WriteUvarint(uint64(prefix))
+	if err != nil {
+		return offset, err
+	}
+
+	_, arrSize := RepeatedUInt32Size(field, v)
+	off, err := s.WriteUvarint(uint64(arrSize))
+	offset += off
+	if err != nil {
+		return offset, err
+	}
+
+	for i := range v {
+		off, err := s.WriteUvarint(uint64(v[i]))
+		offset += off
+		if err != nil {
+			return offset, err
+		}
+	}
+
+	return offset, nil
+}
+
+// RepeatedInt32Marshal implements Stream interface.
+func (s *stream) RepeatedInt32Marshal(field int, v []int32) (int, error) {
+	if len(v) == 0 {
+		return 0, nil
+	}
+
+	prefix := field<<3 | 0x02
+	offset, err := s.WriteUvarint(uint64(prefix))
+	if err != nil {
+		return offset, err
+	}
+
+	_, arrSize := RepeatedInt32Size(field, v)
+	off, err := s.WriteUvarint(uint64(arrSize))
+	offset += off
+	if err != nil {
+		return offset, err
+	}
+
+	for i := range v {
+		off, err := s.WriteUvarint(uint64(v[i]))
+		offset += off
+		if err != nil {
+			return offset, err
+		}
+	}
+
+	return offset, nil
+}
+
+// NestedStructureMarshal implements Stream interface.
+func (s *stream) NestedStructureMarshal(field int64, v StreamMarshaler) (int, error) {
+	if v == nil || reflect.ValueOf(v).IsNil() {
+		return 0, nil
+	}
+
+	var (
+		offset, n int
+		err       error
+	)
+
+	prefix, _ := NestedStructurePrefix(field)
+
+	n, err = s.WriteUvarint(prefix)
+	if err != nil {
+		return n, err
+	}
+
+	offset += n
+
+	n, err = s.WriteUvarint(uint64(v.StableSize()))
+	if err != nil {
+		return offset + n, err
+	}
+
+	offset += n
+	n, err = v.MarshalStream(s)
```

Comment:
`err` ignored there and the same way at:
- v2/netmap/marshal.go:680
- v2/netmap/marshal.go:751:5